smoketests/__init__.py

```diff
@@ -129,9 +129,9 @@ def log_records(self, n):
     def publish_module(self, domain=None, *, clear=True, capture_stderr=True):
         publish_output = self.spacetime(
             "publish",
-            *[domain] if domain is not None else [],
             "--project-path", self.project_path,
-            *(["-c"] if clear else []),
+            *[domain] if domain is not None else [],
+            *["-c"] if clear and domain is not None else [],
```

Comment:
I just changed `if clear` to `if clear and domain is not None`, but I also moved the lines so the `domain is not None`s are side-by-side.